### ISSUE TITLE
Spack won't let me build *anything* if openmpi is mis-configured

### ISSUE THREAD

**Issue by hartzell** [LLNL/spack#4162]
Warning: I know next to nothing about openmpi...

I've been asked to help out on a cluster that has problems.

I thought I'd start out by building some tools.  Anything interesting that I try to do (even fetching) dies as follows:

```
[me@blah spack]$ ./bin/spack fetch gcc@5.4.0
==> Error: Command exited with status 1:
'/usr/bin/ibv_devices'
[me@blah spack]$ /usr/bin/ibv_devices
Failed to get IB devices list: Function not implemented
[me@blah spack]$
```

I tracked it down to the OpenMPI package evaluating the code in `_verbs_dir()`, which involves executing `/usr/bin/ibv_devices` if it exists.

I made this uber-simple hack to disable that bit and I'm now able to do basic stuff (haven't tried anything fancier yet...

```diff
diff --git a/var/spack/repos/builtin/packages/openmpi/package.py b/var/spack/repos/builtin/packages/openmpi/package.py
index 2761df5..6f13ce3 100644
--- a/var/spack/repos/builtin/packages/openmpi/package.py
+++ b/var/spack/repos/builtin/packages/openmpi/package.py
@@ -32,6 +32,7 @@ def _verbs_dir():
     """Try to find the directory where the OpenFabrics verbs package is
     installed. Return None if not found.
     """
+    return None
     try:
         # Try to locate Verbs by looking for a utility in the path
         ibv_devices = which("ibv_devices")
```

This slightly more principled fix also seems to work.

```diff
diff --git a/var/spack/repos/builtin/packages/openmpi/package.py b/var/spack/repos/builtin/packages/openmpi/package.py
index 2761df5..a97584f 100644
--- a/var/spack/repos/builtin/packages/openmpi/package.py
+++ b/var/spack/repos/builtin/packages/openmpi/package.py
@@ -48,6 +48,8 @@ def _verbs_dir():
         return path
     except TypeError:
         return None
+    except ProcessError:
+        return None


 class Openmpi(AutotoolsPackage):
```

Granted, this cluster is SNAFU/FUBAR/..., but should the openmpi package be more robust in the face of this sort of thing?  Is, perhaps, my second solution worthy of a PR?

**Comment by alalazo** [comment]
@hartzell 

> Is, perhaps, my second solution worthy of a PR?

:+1: for me.

**Comment by adamjstewart** [comment]
Closed by #4183
